cdap-security/src/main/java/io/cdap/cdap/security/store/FileSecureStoreImporter.java

```diff
@@ -0,0 +1,188 @@
+    try {
+      try (InputStream in = new DataInputStream(Files.newInputStream(path))) {
+        keyStore.load(in, password);
+      }
+    } catch (Exception e) {
+      attemptUpgrade(keyStore, path, password);
+    }
+  }
+
+  private void attemptUpgrade(KeyStore keyStore, Path path, char[] password) throws IOException,
+    IllegalArgumentException {
+    // Instantiate a new KeyStore.
+    try {
+      keyStore.load(null, password);
+    } catch (NoSuchAlgorithmException | CertificateException e) {
+      throw new IOException("Failed to initialize empty keystore for upgrade:", e);
+    }
+
+    KeyStore historicalKeyStore = null;
+    boolean historicalKeyStoreLoadSuccess = false;
+    // KeyStore files using older schemas may fail to import. In these cases, we attempt to upgrade the schema.
+    for (Class<? extends FileSecureStoreCodec> historicalCodecClass : HISTORICAL_CODECS) {
+      FileSecureStoreCodec historicalCodec;
+      // Instantiate an instance of the historical codec.
+      try {
+        historicalCodec = historicalCodecClass.newInstance();
+      } catch (InstantiationException | IllegalAccessException e) {
+        throw new IllegalArgumentException("Failed to instantiate historical codec:", e);
+      }
+
+      // Instantiate an instance of the historical KeyStore.
+      try {
+        historicalKeyStore = KeyStore.getInstance(historicalCodec.getKeyStoreScheme());
+      } catch (KeyStoreException e) {
+        throw new IllegalArgumentException("Failed to get historical keystore instance:", e);
+      }
+
+      try {
+        // Load the historical KeyStore.
+        try (InputStream in = new DataInputStream(Files.newInputStream(path))) {
+          historicalKeyStore.load(in, password);
+        } catch (NoSuchAlgorithmException | CertificateException e) {
+          LOG.warn("Failed to load historical keystore:", e);
+          continue;
+        }
+
+        historicalKeyStoreLoadSuccess = true;
+
+        // Iterate over all aliases in the keystore, deserialize them using the historical codec, re-serialize them
+        // using the current codec, and import them to the new KeyStore.
+        Enumeration<String> historicalAliases;
+        try {
+          historicalAliases = historicalKeyStore.aliases();
+        } catch (KeyStoreException e) {
+          throw new IOException("Failed to load historical keys:", e);
+        }
+        while (historicalAliases.hasMoreElements()) {
+          String historicalAlias = historicalAliases.nextElement();
+          upgradeKey(keyStore, historicalKeyStore, historicalAlias, historicalCodec, password);
+        }
+      } catch (Exception e) {
+        // If we hit an exception with a particular codec, we continue to try with other codecs.
+        LOG.warn("Failed to upgrade using historical codec {}", historicalCodecClass.getName(), e);
+      }
+
+      // If we get here, the historical keystore load was successful, so break out of the loop.
+      break;
```

Comment:
You can use `return` instead and that saves the need of `historicalKeyStoreLoadSuccess`.

Reply:
Fixed